packages/api/src/routers/api/alerts.ts

```diff
@@ -71,6 +74,57 @@ const validateGroupBy = async (req, res, next) => {
 };
 
 // Routes
+router.get('/', async (req, res, next) => {
+  try {
+    const teamId = req.user?.team;
+    if (teamId == null) {
+      return res.sendStatus(403);
+    }
+    const alerts = await Alert.find({ team: teamId });
+    const alertHistories: any = {};
+    for (const alert of alerts) {
+      const histories = await AlertHistory.find({ alert: alert._id })
+        .sort({ createdAt: -1 })
+        .limit(20);
+      alertHistories[alert._id.toString()] = histories;
+    }
+    res.json({
+      data: {
+        alerts: alerts,
+        histories: alertHistories,
+      },
+    });
+  } catch (e) {
+    next(e);
+  }
+});
+
+router.get('/:id', validateGet, async (req, res, next) => {
+  try {
+    const teamId = req.user?.team;
+    if (teamId == null) {
+      return res.sendStatus(403);
+    }
+    const { id } = req.params;
+    const alert = await Alert.findOne({ _id: id, team: teamId });
+    if (!alert) {
+      return res.sendStatus(404);
+    }
+    const alertHistories = await AlertHistory.find({
+      alert: id,
+      team: teamId,
+    }).limit(20);
```

Comment:
should we sort here ?

Reply:
Yes we should, thank you for catching it. it should be $gte createdAt a week ago and sorted by createdAt desc or something

- [ ] todo fix sorting and filtering on index page